precompiles/werc20/events.go

```diff
@@ -0,0 +1,71 @@
+// Copyright Tharsis Labs Ltd.(Evmos)
+// SPDX-License-Identifier:ENCL-1.0(https://github.com/evmos/evmos/blob/main/LICENSE)
+
+package werc20
+
+import (
+	"math/big"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/vm"
+
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	cmn "github.com/evmos/evmos/v15/precompiles/common"
+)
+
+const (
+	// EventTypeDeposit defines the event type for the Deposit transaction.
+	EventTypeDeposit = "Deposit"
+	// EventTypeWithdraw defines the event type for the Withdraw transaction.
+	EventTypeWithdraw = "Withdraw"
+)
+
+// EmitDepositEvent creates a new Deposit event emitted on a Deposit transaction.
+func (p Precompile) EmitDepositEvent(ctx sdk.Context, stateDB vm.StateDB, dst common.Address, amount *big.Int) error {
+	event := p.Precompile.ABI.Events[EventTypeDeposit]
+	return p.createWERC20Event(ctx, stateDB, event, dst, amount)
+}
+
+// EmitDepositEvent creates a new Withdraw event emitted on Withdraw transaction.
+func (p Precompile) EmitWithdrawEvent(ctx sdk.Context, stateDB vm.StateDB, src common.Address, amount *big.Int) error {
```

Comment:
In other precompiles we don't have two functions: Emit<Event> + create<Event>. What is the advantage of this approach? I would maintain coherence in the code and use only Emit<Event> or this two functions structure.

Reply:
> What is the advantage of this approach?

reduce duplicated code since it's exactly the same event with a different name